Software/scan_key/AuthenticationSQL.py

```diff
@@ -1,42 +1,42 @@
 from on_off import *
 from ErrorSQL import *
 import os
+import subprocess
 
 def machineAuth(id, cursor):
-    os.system("hostname > tmp")
-    currMachineID = open('tmp', 'r').read()
-    currMachineID = currMachineID.split()
+    currMachineID = subprocess.check_output(['hostname'])
+    name = name.strip()
 
-    cursor.execute("select machineID, machineType from MACHINE")
+    cursor.execute("select machineID, machineType FROM MACHINE WHERE t1String = " + id)
 
-    data1 = cursor.fetchall()
+    data1 = cursor.fetchone()
 
     machineType = None
-
+    
     for origMachineID in data1:
-        if(origMachineID[0] == currMachineID[0]):
+        if(origMachineID[0] == currMachineID):
             machineType = origMachineID[1]
 
     cursor.execute("SELECT auth1, auth2, auth3 FROM USER WHERE t1String = " + id) #getting user w/ the id
 
-    authdata = cursor.fetchall()
+    authdata = cursor.fetchone()
 
     for auth in authdata:
         if(auth[machineType] != 1):
             print ("USER IS NOT AUTHORIZED\n")
+            os.system("omxplayer deny.wav &")
             errorSQL(id, 1)
             return (True)
```

Comment:
Wait a minute - none of this (24-29) belongs in this function!

It reads like this function is simply supposed return the machine authentication type. We should try to avoid duplication of logic.